activiti-cloud-services-runtime-bundle/activiti-cloud-services-job-executor/src/main/java/org/activiti/cloud/services/job/executor/MessageBasedJobManagerConfigurator.java

```diff
@@ -0,0 +1,151 @@
+    
+    public MessageBasedJobManagerConfigurator(ConfigurableListableBeanFactory beanFactory,
+                                              BindingService bindingService,
+                                              JobMessageInputChannelFactory inputChannelFactory,
+                                              MessageBasedJobManagerFactory messageBasedJobManagerFactory,
+                                              JobMessageHandlerFactory jobMessageHandlerFactory,
+                                              ConsumerProperties consumerProperties) {
+        this.bindingService = bindingService;
+        this.inputChannelFactory = inputChannelFactory;
+        this.consumerProperties = consumerProperties;
+        this.messageBasedJobManagerFactory = messageBasedJobManagerFactory;
+        this.jobMessageHandlerFactory = jobMessageHandlerFactory;
+        this.beanFactory = beanFactory;
+    }
+    
+    protected MessageHandler createJobMessageHandler(ProcessEngineConfigurationImpl configuration) {
+        MessageHandler messageHandler = jobMessageHandlerFactory.create(configuration);
+        
+        beanFactory.registerSingleton(JOB_MESSAGE_HANDLER, messageHandler);
+
+        return (MessageHandler) beanFactory.initializeBean(messageHandler, JOB_MESSAGE_HANDLER);
+    }
+    
+    /**
+     * Configures MessageBasedJobManager 
+     */
+    @Override
+    public void beforeInit(ProcessEngineConfigurationImpl configuration) {
+        
+        messageBasedJobManager = messageBasedJobManagerFactory.create(configuration);
+        
+        // Let's manage async executor lifecycle manually on start/stop
+        configuration.setAsyncExecutorActivate(false);
+        configuration.setAsyncExecutorMessageQueueMode(true);
+        configuration.setJobManager(messageBasedJobManager);
+    }
+
+    /**
+     * Configures input channel 
+     */
+    @Override
+    public void configure(ProcessEngineConfigurationImpl configuration) {
+        this.configuration = configuration;
+        
+        String channelName = messageBasedJobManager.getInputChannelName();
+        String destination = messageBasedJobManager.getDestination();
+        
+        BindingProperties bindingProperties = new BindingProperties();
+        bindingProperties.setConsumer(consumerProperties);
+        bindingProperties.setContentType(APPLICATION_JSON);
+        bindingProperties.setGroup(JOB_MESSAGE_HANDLER);
+        // Let's use message job producer destination scope
+        bindingProperties.setDestination(destination);
+
+        // Let's create input channel 
+        inputChannel = inputChannelFactory.createInputChannel(channelName, bindingProperties);
+    }
+
+    @Override
+    public int getPriority() {
+        return 0;
+    }
+
+    @Override
+    public void start() {
+        jobMessageHandler = createJobMessageHandler(configuration);
+        
+        // Let's subscribe and bind consumer channel   
```

Comment:
It is a part of start/stop unit of work. I think it will only make sense to externalize it if we allow to override configuration behavior. My opinion we should not allow to override it without need.

Reply:
I agree that we should not allow override, but the method is can be private (so not overridable). I'm just concerned about inline comments: if you few the need of adding a comment to describe what a couple of lines is doing, probably you need a (private) method for that.